spec/controllers/alchemy/admin/elements_controller_spec.rb

```diff
@@ -13,31 +20,27 @@ module Alchemy
       let(:alchemy_page) { build_stubbed(:alchemy_page) }
 
       before do
-        expect(Page).to receive(:find).and_return alchemy_page
+        expect(Page).to receive(:find_by).and_return alchemy_page
       end
 
       context 'with cells' do
-        let(:cell) { build_stubbed(:alchemy_cell, page: alchemy_page) }
+        let(:cell) { create(:alchemy_cell, page: alchemy_page) }
 
         before do
-          expect(alchemy_page).to receive(:cells).and_return [cell]
+          alchemy_page.cells << cell
         end
 
         it "groups elements by cell" do
           expect(alchemy_page).to receive(:elements_grouped_by_cells)
-          alchemy_get :index, {page_id: alchemy_page.id}
+          alchemy_get :index, {page_version_id: alchemy_page.current_version_id}
```

Comment:
Here you're changing the API: I think it's better to pass the Page ID and have the controller/model figure out that it really wants to edit the current version.